lib/rex/proto/ssh/connection.rb

```diff
@@ -83,15 +83,16 @@ def initialize(io = nil, options = self.default_options, context = {})
     @context = context
     @logger = Logger.new self.class.name
     @server = options.delete(:ssh_server)
+    @mode = options.delete(:ssh_mode) || HrrRbSsh::Mode::SERVER
```

Comment:
The reverse ssh handler worked with hrr_rb_ssh v0.3.0.pre2 - but the upstream ssh library changed their API in in v0.3.0 to require the client/server mode being set on the connection object, and authentication objects as well as the transport object

https://github.com/hirura/hrr_rb_ssh/commit/98f72910bdafb1cf2a8feaf993ff1db5b9458c55

When we bumped to hrr_rb_ssh-ed25519 as part of https://github.com/rapid7/metasploit-framework/pull/15493/commits/ab31500d7f0c413a93eacd713b39b3e694114b04 - it grabbed the latest version of `hrr_rb_ssh` - which is version 0.4.2 which caused things to break